### DISCUSSION TITLE
networked-audio-using-vban-and-rpi

### DISCUSSION THREAD

**Discussion by giscus** [engineerjoe440/stanley-solutions-blog#7] [Comments]
# networked-audio-using-vban-and-rpi

Ever wish your music was synchronized across your home, but you didn't have to sell your soul to Google to make it happen? Well, I did it with a Raspberry Pi and a nifty little open-source project!

https://blog.stanleysolutionsnw.com/networked-audio-using-vban-and-rpi.html

**Comment by Paul-vk3tgx** [comment]
Interesting, looking for a minimalist 'thing' to receive VB audio, that hopefully does not mind being just switched on and off as one would do with regular audio kit.

**Reply by engineerjoe440** [reply]
Definitely, @Paul-vk3tgx! There's some other parts of this that make it a bit challenging, for sure. I'd still like to improve the overall behavior, but it's been decent.

**Comment by davidinfosec** [comment]
Excellent post! I followed along, and everything went wonderful. I plan to implement something like this in a corporate environment with a church for sending our main audio feed throughout rooms locally over WiFi. Thank you!

**Comment by RobNO97** [comment]
Hi there,

I tried it just before a fresh install, and it was working flawlessly.
I then had to reinstall my OS, and now I have crackling audio on high sample rate and low buffer (44100Hz on voicemeeter and -q 1 setting gives me "short write" error :
Error: alsa_write: short write (expected 256, wrote 242)
Error : alsa_write : snd_pcm__writei failed : broken pipe
ALSA lib pcm.c:8570:(snd_pcm_recover) underrun occured

I can't find anything on the web...and it was working before with the same settings, evend working with -q 0 and 48000Hz.

Any idea ?

**Reply by engineerjoe440** [reply]
Hey there, @RobNO97!

I've run into that, only about a few hundred thousand times. Hah! I ended up wrapping the script on the side of my Raspberry Pi to observe for any such failures, and restart the script. You can take a look at that script in another one of my blog posts:

https://blog.stanleysolutionsnw.com/spam-the-vban-for-non-stop-audio.html

I'll say that it's not ideal, by any means, and my script is a bit hacky. Hah!

---

To address your comment about 48kHz working, I've recently learned that modern Linux systems using PipeWire default to 48000Hz, so if that's working for you, I'd personally be inclined to run with it.

Admittedly, if I were to rebuild everything all over again, I'd certainly look at doing everything at 48kHz. I wouldn't doubt that at some point in my future, I'll end up doing that, just due to my own tendency to accidentally "break" things. Hah!

**Reply by RobNO97** [reply]
Sorry maybe it was unclear : I don't need to spam VBAN as it's already working, up and running. But the sound quality is not here.
In fact, if I use a too small buffer (-q 0 or -q1), audio is crackling. the pipe is still there and working but it's generating the above errors indefinitely as the audio is crackling (several time per seconds).

Crackling audio is the symptoms of the generated errors in real time

**Reply by engineerjoe440** [reply]
That's a pretty accurate description of what errors I encountered occasionally. My resolution was to use the "wrapper script" to identify when they began to occur and restart the VBAN client. Hope that helps! :smile:
